columnar/src/columnar/merge/mod.rs

```diff
@@ -287,10 +289,44 @@ fn merged_numerical_columns_type<'a>(
     compatible_numerical_types.to_numerical_type()
 }
 
+fn is_empty_after_merge(
+    merge_row_order: &MergeRowOrder,
+    column: &DynamicColumn,
+    columnar_id: usize,
+) -> bool {
+    if column.num_values() == 0u32 {
+        // It was empty before the merge.
+        return true;
+    }
+    match merge_row_order {
+        MergeRowOrder::Stack(_) => {
+            // If we are stacking the columnar, no rows are being deleted.
+            false
+        }
+        MergeRowOrder::Shuffled(shuffled) => {
+            if let Some(alive_bitset) = &shuffled.alive_bitsets[columnar_id] {
+                let column_index = column.column_index();
+                for doc in alive_bitset.iter() {
+                    if column_index.has_value(doc) {
+                        return false;
+                    }
+                }
```

Comment:
Similar like above, iterating the other way around should be much faster, especially in cases with many sparse columns

but here, it may take much longer to hit the early exit for very sparse columns

Pathological case:  10mio docs, 100k Columns with each one value. Average has_value early exit after 5mio

100k * 5mio = `500_000_000_000` calls to `has_value` (that's 8minutes for 1ns per call)

Reply:
I made the change.